frontend/src/components/countdown.js

```diff
@@ -1,53 +1,65 @@
 import React, { Component } from "react";
+import { NavLink } from "react-router-dom";
 import Api from "../api";
 import Duration from "duration";
+import { getDateTimeText } from "../utils/date";
 
+import Spinner from "./spinner";
 class Countdown extends Component {
   constructor() {
     super();
 
     // Initialize defaults to render before API call finishes
     // Make sure this is as minimal as possible
     this.state = {
-      days: 0,
-      hours: 0,
-      min: 0,
-      sec: 0,
+      days: null,
+      hours: null,
+      min: null,
+      sec: null,
     };
   }
 
   componentDidMount() {
-    Api.getNextTournament((tournament_info) => {
-      // Note that state updates asynchronously, but we need it immediately for computation.
-      // So we can't only set state here
-      this.submission_deadline = tournament_info.submission_deadline;
-      this.setState({ submission_deadline: this.submission_deadline });
-      const text_submission_deadline = tournament_info.submission_deadline_strs;
-      this.setState({ text_submission_deadline });
-
-      this.refreshCountdown();
-      this.interval = setInterval(() => {
-        this.refreshCountdown();
-      }, 1000);
-
-      this.setState({ tournament_name: tournament_info.tournament_name });
-      this.setState({
-        has_next_tournament: tournament_info.has_next_tournament,
+    if (this.props.is_game_released) {
+      Api.getNextTournament(this.props.episode, (next_tournament) => {
+        this.setState({ next_tournament });
+        if (next_tournament) {
+          this.setState({
+            countdown_time: new Date(next_tournament.submission_freeze),
+          });
+        }
+        //   const has_next_tournament = next_tournament !== null;
+        //   this.setState({has_next_tournament});
+
+        //   if (has_next_tournament) {
+        //     this.setState({ countdown_time: new Date(next_tournament.submission_freeze) });
+
+        //     this.setState({ tournament_name: next_tournament.tournament_name });
+        //     this.setState({
+        //       does_tour_require_resume: next_tournament.does_tour_require_resume,
+        //     });
+        //   }
       });
-    });
+    } else {
+      this.setState({ countdown_time: new Date(this.props.game_release) });
+    }
+    this.refreshCountdown();
+    this.interval = setInterval(() => {
+      this.refreshCountdown();
+    }, 1000);
```

Comment:
This doesn't work in a live-test, I always get a blank countdown.

I think it's caused by a "race" (not really): the callback in lines 27-29 is async. So, these lines happen first, and the interval is set before the actual countdown time is known. `this.refreshCountdown` therefore calls `this.stop()` and kills the countdown before the callback runs.

I think the fix is to move these lines into the callback?